servicetest/filesystem/test_filesystem.py

```diff
@@ -251,15 +259,119 @@ def test_tmpfs_writebuffer_size_enabled(self):
+                    interesting_partition = part
+                    break
+            assert interesting_partition is not False
+
+            # Write a 9 MB file and make sure it is in the right size area (due
+            # to how filesystems works, it may be a bit bigger or smaller so
+            # exact comparison is not possible)
+            with open(absoluteTestFile, "w") as f:
+                f.write("w" * 9 * 1024 * 1024)
+
+            # As mentioned exact comparison is not possible, so approximately
+            # right at least....
+            assert os.path.getsize(absoluteTestFile) >= 8 * 1024 * 1024
+            assert os.path.getsize(absoluteTestFile) <= 10 * 1024 * 1024
+
+            # Try to write a file that is too big and it should raise an
+            # IOError exception.
+            with open(absoluteTestFile, "w") as f:
+                with pytest.raises(IOError):
+                    f.write("w" * 105 * 1024 * 1024)
+        finally:
+            ca.terminate()
+
+        if os.path.exists(absoluteTestFile):
+            os.remove(absoluteTestFile)
+
+    @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
+    def test_tmpfs_writebuffer_size_missing_size_config(self):
+        """ This test will test various bad configuration combinations and
+            make sure that SoftwareContainerAgent behaves as expected when
+            the bad configuration is sent to it.
+
+            This tests that the container is created even if the size parameter
+            is missing from the configuration.
+        """
+
+        absoluteTestFile = os.path.join("/tmp/container/SC-0/rootfs-upper",
+                                        TESTFILE)
+
+        ca = Container()
+
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true, \
+                               "enableTemporaryFileSystemWriteBuffer": true}]'
+
+        try:
+            ca.start(DATA)
+        finally:
+            ca.terminate()
+
+        if os.path.exists(absoluteTestFile):
+            os.remove(absoluteTestFile)
+
+    @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
+    def test_tmpfs_writebuffer_size_missing_tmpfs_enable_flag(self):
+        """ This test will test various bad configuration combinations and
+            make sure that SoftwareContainerAgent behaves as expected when
+            the bad configuration is sent to it.
+
+            This run should work, but no temporary filesystem will be setup as
+            the enableTemporaryFileSystemWriteBuffer defaults to false and no
+            parsing will be performed on the temporaryFileSystemSize parameter.
+        """
+
+        absoluteTestFile = os.path.join("/tmp/container/SC-0/rootfs-upper",
+                                        TESTFILE)
+
+        ca = Container()
+
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true, \
+                               "temporaryFileSystemSize": ' + str(TenMB) + '}]'
+
+        try:
+            ca.start(DATA)
```

Comment:
Will this also verify that it was successful? Will this fail if it is not possible?